tests/iscsi/iscsi_client.pm

```diff
@@ -16,47 +16,139 @@ use strict;
+    wait_screen_change { type_string $initiator_conf{acl_pass}; };
+    save_screenshot;
+    send_key 'alt-n';
+    save_screenshot;
+}
+
+sub initiator_connected_targets_tab {
+    send_key "alt-d";    # go to discovered targets tab
     assert_screen 'iscsi-initiator-discovered-targets';
-    send_key "alt-n";                                     # next
+    send_key "alt-n";    # go to connected targets tab
     assert_screen 'iscsi-initiator-connected-targets';
-    send_key "alt-o";                                     # OK
-    wait_still_screen(2, 10);
+    send_key "alt-o";    # OK
+}
+
+
+sub run {
+    prepare_xterm;
+    record_info 'Network', 'Configure MM network - client';
+    configure_static_mm_network(network_address => $initiator_conf{ip});
+    mutex_wait('iscsi_target_ready');    # wait for server setup
+    record_info 'Target Ready!', 'iSCSI target is configured, start initiator configuration';
+    type_string "yast2 iscsi-client; echo yast2-iscsi-client-\$? > /dev/$serialdev\n";
+    initiator_service_tab;
+    initiator_discovered_targets_tab;
+    initiator_connected_targets_tab;
+    wait_serial("yast2-iscsi-client-0", 180) || die "'yast2 iscsi-client ' didn't finish or exited with non-zero code";
+    # logging in to a target will create a local disc device
+    # it takes a moment, since udev actually handles it
+    sleep 5;
+    record_info 'Systemd', 'Verify status of iscsi services and sockets';
+    systemctl("is-active iscsid.service",    expect_false => 0);
+    systemctl("is-active iscsid.socket",     expect_false => 0);
+    systemctl("is-active iscsi.service",     expect_false => 0);
+    systemctl("is-active iscsiuio.service",  expect_false => 0);
+    systemctl("is-active iscsiuio.socket",   expect_false => 1);
+    systemctl("is-active targetcli.service", expect_false => 1);
+    record_info 'Verify LUN availability';
     assert_script_run 'lsscsi';
+    # later delete, moved to post_fail hook
+    assert_script_run 'iscsiadm --mode node session -P 1';
+    assert_script_run 'ls /dev/disk/by-path';
+    assert_script_run 'lsblk --scsi';
+    # making a single partition actually causes the kernel code to re-read the starting part of the disc
+    # in order for it to recognize that you now have a partition table when before there was none
     assert_script_run "echo -e \"n\\np\\n1\\n\\n\\nw\\n\" \| fdisk /dev/sda";    # create one partition
+    sleep 2;
     assert_script_run 'mkfs.ext4 /dev/sda1';                                     # format partition to ext4
+    sleep 2;
     assert_script_run 'mount /dev/sda1 /mnt';                                    # mount partition to /mnt
     assert_script_run 'echo "iscsi is working" > /mnt/iscsi';                    # write text to file on iscsi disk
     assert_script_run 'grep "iscsi is working" /mnt/iscsi';                      # grep expected text from file
+    mutex_create('iscsi_initiator_ready');
+    mutex_wait('iscsi_display_sessions');
```

Comment:
Consider adding some comments to mutexes - makes it easier when you look at it year later :)
For example: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/caasp.pm#L239